### PR TITLE
[6.x] Core/PacketIO: Update and enable most pet packets

### PR DESCRIPTION
**Pull request by myran2**

### CODE REVIEW

File: src/server/game/Server/Packets/PetPackets.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+WorldPacket const* WorldPackets::Pets::PetNameInvalid::Write()
+{
+    _worldPacket << RenameData.PetGUID;
+    _worldPacket << int32(RenameData.PetNumber);
+
+    _worldPacket << uint8(RenameData.NewName.length());
+
+    _worldPacket.WriteBit(RenameData.HasDeclinedNames);
```
Comment (MitchesD) [inline src/server/game/Server/Packets/PetPackets.cpp]: flush bits after writebit

File: src/server/game/Server/Packets/PetPackets.cpp [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    if (RenameData.HasDeclinedNames)
+    {
+        for (int i = 0; i < MAX_DECLINED_NAME_CASES; i++)
+            _worldPacket.WriteBits(RenameData.DeclinedNames.name[i].length(), 7);
```
Comment (MitchesD) [inline src/server/game/Server/Packets/PetPackets.cpp]: there you should flush too

File: src/server/game/Server/Packets/PetPackets.cpp [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    _worldPacket >> nameLen;
+
+    RenameData.HasDeclinedNames = _worldPacket.ReadBit();
+    if (RenameData.HasDeclinedNames)
+    {
+        int count[MAX_DECLINED_NAME_CASES];
+        for (int i = 0; i < MAX_DECLINED_NAME_CASES; i++)
+        {
+            count[i] = _worldPacket.ReadBits(7);
+            _worldPacket.FlushBits();
```
Comment (MitchesD) [inline src/server/game/Server/Packets/PetPackets.cpp]: useless, write only usage

File: src/server/game/Server/Packets/PetPackets.cpp [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        }
+
+        for (int i = 0; i < MAX_DECLINED_NAME_CASES; i++)
+        {
+            RenameData.DeclinedNames.name[i] = _worldPacket.ReadString(count[i]);
+            _worldPacket.FlushBits();
```
Comment (MitchesD) [inline src/server/game/Server/Packets/PetPackets.cpp]: remove it

**Comment by msoky** [comment]
:+1:

**Comment by tkrokli** [comment]
Maybe copy the original PR text from https://github.com/TrinityCore/TrinityCore/pull/16028 and paste it in the PR comment?

**Comment by sarnex** [comment]
@tkrokli shigeNO

**Comment by tkrokli** [comment]
Waarudo izu. 生活し、学びます / 學無止境

**Comment by sarnex** [comment]
@tkrokli PJSalt ?

**Comment by tkrokli** [comment]
Kappa.

**Comment by jackpoz** [comment]
> No description provided.

It's a bad habit to open PRs with no descriptions

**Comment by myran2** [comment]
Should I change the namespace from Pets to Pet to coincide with @Carbenium's changes?

**Comment by Shauren** [comment]
yes

**Comment by Carbenium** [comment]
Ah sorry @myran2. Didn't realize that there was a PR open.

**Comment by tkrokli** [comment]
The branch `myran2:pets` needs to be synchronized with the TC 6.x branch (rebase).

**Comment by myran2** [comment]
Yep. I'll get on that soon.

**Comment by DDuarte** [comment]
Conflicts solved and merged in 67b977c86a0d5fcbe5b3c89e9a19275b775d31f7

Thanks for the PR :+1:

**Pull request closed without merging**